### PR TITLE
Blog - Using CMake and GCC to Cross-Compile Binaries

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@balapriyac [Preview Link](https://645a8aed0cf584157701cb79--ephemeral-narwhal-3ac3b2.netlify.app/blog/cmake-gcc-cross-compile/)

**Comment by msali123** [comment]
[Updated link](https://645b7af49f2beb0fd99c5414--ephemeral-narwhal-3ac3b2.netlify.app/blog/cmake-gcc-cross-compile/)

File: blog/_posts/2023-06-16-cmake-gcc-cross-compile.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+internal-links:
+ - CMake
+ - GCC
+ - Cross-Compile Binaries
+ - Cross-compilation
+ - C++
+---
+
+**We're [Earthly](https://earthly.dev/). We make building software simpler—and faster—using containerization. This article covers cross-compiling binaries using CMake and GCC. If you want to see what can be done by combining ideas from a `Makefile` and a `Dockerfile` then [check us out](/).**
```
Comment (balapriyac) [inline blog/_posts/2023-06-16-cmake-gcc-cross-compile.md:17]:
```suggestion
**We're [Earthly](https://earthly.dev/). We make building software simpler—and faster—using containerization. This article covers cross-compiling binaries using CMake and GCC. If you want to see what can be done by combining ideas from a `Makefile` and a `Dockerfile`, then [check us out](/).**
```

File: blog/_posts/2023-06-16-cmake-gcc-cross-compile.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+sudo dnf groupinstall 'Development Tools'
+~~~
+
+GNU make comes preinstalled on most Linux distributions. You can check if it's installed on your machine using the following command:
+
+~~~{.bash caption=">_"}
+make --version
+~~~
+
+If [make](/blog/makefiles-on-windows) is not installed, use the following command to install it on Debian/Ubuntu:
```
Comment (balapriyac) [inline blog/_posts/2023-06-16-cmake-gcc-cross-compile.md:49]:
```suggestion
If make is not installed, use the following command to install it on Debian/Ubuntu:
```

File: blog/_posts/2023-06-16-cmake-gcc-cross-compile.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+</div>
+
+#### Compiling With Make
+
+Now that you've compiled your program with GCC, compile the program with Make, a popular build automation tool that provides granular control over the build process by defining dependencies between files and targets. In doing so, Make can determine the correct order of operations needed to build a project efficiently.
+
+The working procedure of Make revolves around a simple text file called the Makefile. It contains rules that tell Make how to build the project from scratch:
+
+<div class="wide">
+![Make diagram]({{site.images}}{{page.slug}}/SshQwn3.png)
```
Comment (balapriyac) [inline blog/_posts/2023-06-16-cmake-gcc-cross-compile.md:109]:
```suggestion
![Make diagram]({{site.images}}{{page.slug}}/SshQwn3.png)\
```

File: blog/_posts/2023-06-16-cmake-gcc-cross-compile.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+Make goes through the Makefile and compiles the `hello.cpp` program using the specified g++ command. Next, delete the generated executable:
+
+~~~{.bash caption=">_"}
+make clean
+~~~
+
+## Cross-Compiling With CMake and GCC
+
+Now that you have a simple C++ program running, it's time to cross-compile it to run on a different architecture using CMake. CMake is a popular tool for managing the build process of C and C++ projects, and it has built-in support for cross-compiling.
```
Comment (balapriyac) [inline blog/_posts/2023-06-16-cmake-gcc-cross-compile.md:141]:
```suggestion
Now that you have a simple C++ program running, it's time to cross-compile it to run on a different architecture using CMake. [CMake](/blog/using-cmake/) is a popular tool for managing the build process of C and C++ projects, and it has built-in support for cross-compiling.
```

File: blog/_posts/2023-06-16-cmake-gcc-cross-compile.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+As you can see, this cross-compiled `hello` executable is running as expected on an ARM64 VM.
+
+## Troubleshooting Common Issues During Cross-Compiling
+
+![Troubleshooting]({{site.images}}{{page.slug}}/troubleshoot.png)\
+
+Even with the help of tools like CMake and GCC, cross-compiling C++ programs can be challenging. During cross-compiling, common issues you may encounter include compiler incompatibility, lack of readily available libraries, and toolchain conflicts.
+
+If you're facing some of these issues, here are a few tips:
+
```
Comment (balapriyac) [inline blog/_posts/2023-06-16-cmake-gcc-cross-compile.md:270]:
```suggestion

Even with the help of tools like CMake and GCC, cross-compiling C++ programs can be challenging. 

During cross-compiling, common issues you may encounter include compiler incompatibility, lack of readily available libraries, and toolchain conflicts. If you're facing some of these issues, here are a few tips:

```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer. Requested a few minor changes.

**Comment by msali123** [comment]
Thanks Bala [Changes done](https://64633c783349905e47f86906--ephemeral-narwhal-3ac3b2.netlify.app/blog/cmake-gcc-cross-compile/)

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Review by jalletto — APPROVED** [review]

**Pull request merged**